include/rovio/PoseUpdate.hpp

```diff
@@ -60,8 +60,10 @@ class PoseUpdateMeas: public LWF::State<LWF::VectorElement<3>,LWF::QuaternionEle
  public:
   static constexpr unsigned int _pos = 0;
   static constexpr unsigned int _att = _pos+1;
+  M3D pos_cov_; // Will be used to scale the update covariance according to the measurement
```

Comment:
The 6x6 matrix would also consider cross-correlations between position and attitude. But I see the problem with the potentially invalid data. What about making the reading in the callback (RovioNode.hpp) dependent on enablePosition_/enableAttitude_?